### PR TITLE
Memory aligned allocation

### PR DESCRIPTION
**Pull request by kouamdo**
PR for aligned memory allocation

### CODE REVIEW

File: src/tests/stdlib/aligned_alloc_test.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}
+TEST_CASE("tEST WITH LARGE SIZE") {
+    test_assert_not_equal(aligned_alloc(64, 0xF00), NULL);
+}
+TEST_CASE("TEST WITH MULTIPLE ALLOCATION") {
+    void *mem[_const_size_];
+    uint32_t i = 0;
+    for (i;i < _const_size_;i++) {
+        mem[i] = aligned_alloc(i, i*0xF00);
+    }
```
Comment (alexkalmuk) [inline src/tests/stdlib/aligned_alloc_test.c:34]: Try to or 
```
for (i = 0; i < _const_size_;i++) {
```

or just

```
for (;i < _const_size_;i++) {
```

Comment (alexkalmuk) [reply src/tests/stdlib/aligned_alloc_test.c:34]: It's related to the compilation error on Travis.

Comment (kouamdo) [reply src/tests/stdlib/aligned_alloc_test.c:34]: Okay .... met me change it....

**Comment by alexkalmuk** [comment]
@kouamdo Build is still broken, but now at another place - https://travis-ci.org/github/embox/embox/jobs/708717202#L1801

```
src/tests/mem/pvalloc_test.c:13:20: fatal error: malloc.h: No such file or directory
```

File: src/compat/linux/include/malloc.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#ifndef MALLOC_H
+
+	#define MALLOC_H
+
+	#include <stdio.h>
+	#include <stddef.h>
+
+	extern void *pvalloc(size_t __size);
+
+#endif	/*	MALLOC_H	*/
```
Comment (alexkalmuk) [inline src/compat/linux/include/malloc.h:18]: Why tabs? Fix formatting

File: src/compat/posix/posix_memalign.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        return 0 ;
+    } else {
+        return 1 ;
+    }
+}
+int posix_memalign (void **__memptr, size_t __alignment, size_t __size) {
+    if (power_of_two(__alignment) != 0 || __size == 0){
+        return EINVAL;
+    }
+    *__memptr = memalign(__alignment,  __size) ;
```
Comment (alexkalmuk) [inline src/compat/posix/posix_memalign.c:28]: Why you are using such `__`?

File: src/compat/posix/posix_memalign.c [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    * The address of the allocated memory will be a multiple of __alignment, which must be a power of two and a multiple of sizeof(void *).  
+    * If __size is 0, then the value placed in  *__memptr is either NULL, or a unique pointer value that can later be successfully passed to free().
+*/
+int power_of_two(int nmber) {
+    if ((nmber & (nmber - 1 )) == 0) {
+        return 0 ;
+    } else {
+        return 1 ;
+    }
+}
```
Comment (alexkalmuk) [inline src/compat/posix/posix_memalign.c:23]: static func?

Comment (alexkalmuk) [reply src/compat/posix/posix_memalign.c:23]: Also it can be just `return (x & (x - 1)) == 0;`

File: src/tests/mem/pvalloc_test.c [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    for (;i < CONST_TEST;i++) {
+        mem[i] = pvalloc(0xF00);
+    }
+    for (i = 0;i < CONST_TEST;i++) {
+        test_assert_not_equal(mem[i], NULL);
+    }
+    for (i = 0;i < CONST_TEST;i++) {
+        free(mem[i]);
+    }
+}
```
Comment (alexkalmuk) [inline src/tests/mem/pvalloc_test.c:37]: spaces. formatting

File: src/tests/stdlib/posix_memalign_test.c [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    void *mem = NULL ;
+    test_assert_equal(posix_memalign(&mem, 64, 0xF00), 0);
+    free(mem);
+}
+TEST_CASE("call a multiple allocation with little size") {
+    void *mem[CONST_TEST];
+    int i = 0;
+    for(;i < CONST_TEST;i++) {
+        posix_memalign(mem[i], 64, 0xF00);
+    }
```
Comment (alexkalmuk) [inline src/tests/stdlib/posix_memalign_test.c:51]: It's incorrect. You should `void **memptr` 

```
int posix_memalign(void **memptr, size_t alignment, size_t size);
```

File: src/tests/mem/pvalloc_test.c (hunk 5 above)
Comment (alexkalmuk) [reply src/tests/mem/pvalloc_test.c:37]: here and everywhere

File: src/compat/linux/include/malloc.h (hunk 2 above)
Comment (kouamdo) [reply src/compat/linux/include/malloc.h:18]: okay ,

File: src/compat/posix/posix_memalign.c (hunk 3 above)
Comment (kouamdo) [reply src/compat/posix/posix_memalign.c:28]: i have looked from the the linux manual , sorry i will remove it

File: src/compat/posix/posix_memalign.c (hunk 4 above)
Comment (kouamdo) [reply src/compat/posix/posix_memalign.c:23]: okay ,

File: src/tests/mem/pvalloc_test.c (hunk 5 above)
Comment (kouamdo) [reply src/tests/mem/pvalloc_test.c:37]: okay , i'm fixing it

File: src/tests/stdlib/posix_memalign_test.c (hunk 6 above)
Comment (kouamdo) [reply src/tests/stdlib/posix_memalign_test.c:51]: how ?? 
please ..

thank you

File: src/compat/linux/include/malloc.h [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+ * @file
+ * @brief
+ *
+ * @author  KOUAM Ledoux
+ * @date    21.07.2020
+ */
+
+#ifndef MALLOC_H
+
+	#define MALLOC_H
```
Comment (anton-bondarev) [inline src/compat/linux/include/malloc.h:11]: Why do you use tabs in these plasese?

File: src/compat/linux/include/malloc.h (hunk 2 above)
Comment (anton-bondarev) [reply src/compat/linux/include/malloc.h:18]: Add e new line please

File: third-party/lib/dlmalloc/malloc.h [hunk 8]
```diff
@@ -97,6 +97,15 @@ struct mallinfo {
 #endif  /* HAVE_USR_INCLUDE_MALLOC_H */
 #endif  /* !NO_MALLINFO */
 
+/*
```
Comment (anton-bondarev) [inline third-party/lib/dlmalloc/malloc.h:100]: Why did you add this?

**Comment by anton-bondarev** [comment]
I've closed PR if you want to contribute these changes please open a new PR
